### PR TITLE
Maxlength and performance

### PR DESCRIPTION
**Pull request by leahjlou**
Support input maxlength, improve performance slightly

### CODE REVIEW

File: src/multi-selector.js [hunk 1]
```diff
@@ -122,17 +133,8 @@ const MultiSelector = React.createClass({
 	},
 
 	inputChange: function(newVal) {
-		const filterItems = this.getFilterItems(this.props.items);
-		const activeIndex = this.state.activeIndex;
-
 		this.props.onInputChange && this.props.onInputChange(e.currentTarget.value);
 
-		if (isNull(activeIndex) && filterItems.length !== 0) {
```
Comment (leahjlou) [inline src/multi-selector.js:130]: Turns out this is unnecessary if we just check if `filterItems` has stuff before we try to access the `activeIndex` item of it. (line 166) That means we don't need to get the filter items here

File: src/multi-selector.js [hunk 2]
```diff
@@ -77,6 +78,15 @@ const MultiSelector = React.createClass({
 		})
 	},
 
+	shouldComponentUpdate: function(nextProps, nextState) {
+		return nextState.searchValue !== this.state.searchValue
+			|| nextProps.initialSelectedItems !== this.state.selectedItems
+			|| nextProps.items !== this.props.items
+			|| nextState.dialogDisplayed !== this.state.dialogDisplayed
+			|| nextState.activeIndex !== this.state.activeIndex
+			|| nextState.mouseIndex !== this.state.mouseIndex;
```
Comment (internettrans) [inline src/multi-selector.js:87]: idk if this is a dumb suggestion, but could you just do `nextState !== this.state && nextProps !== this.props`?

Comment (leahjlou) [inline src/multi-selector.js:87]: I was kinda under the impression that if I did that there would be no difference from the default way React determines whether to update. So I tried to only check the values we really care about? It's highly possible i'm going about this the wrong way entirely

Comment (internettrans) [inline src/multi-selector.js:87]: I'm not sure what the default behavior is for shouldComponentUpdate when dealing with Classes (as opposed to stateless function components). What you have here is definitely better than no shouldComponentUpdate at all.

File: src/multi-selector.js [hunk 3]
```diff
@@ -308,14 +310,16 @@ const MultiSelector = React.createClass({
 
 		if (this.state.dialogDisplayed) {
 			let placeholder = this.props.placeholder ? this.props.placeholder : "Type a collaborators name...";
+			let maxLength = this.props.maxLength ? this.props.maxLength : '';
```
Comment (dckesler) [inline src/multi-selector.js]: maxLength is either a number or an empty string?

File: src/multi-selector.js (hunk 2 above)
Comment (dckesler) [inline src/multi-selector.js:87]: I believe Leah is right but I'm not 100%

**Comment by dckesler** [comment]
:cactus:

**Pull request merged**